doc/functions/ocitools.xml

```diff
@@ -0,0 +1,101 @@
+  <link xlink:href="https://github.com/opencontainers/runtime-spec">OCI
+  container specification v1.0.0</link>. Beyond that it makes no assumptions
+  about the container runner you choose to use to run the created container.
+ </para>
+
+ <section xml:id="ssec-pkgs-ociTools-buildContainer">
+  <title>buildContainer</title>
+
+  <para>
+   This function creates a simple OCI container that runs a single command
+   inside of it. An OCI container consists of a <varname>conig.json</varname>
+   and a rootfs directory. The provided command will provide all required
+   dependencies from the nix-store in the container.
+  </para>
+
+  <para>
+   The parameters of <varname>buildContainer</varname> with an example value
+   are described below:
+  </para>
+
+  <example xml:id='ex-ociTools-buildContainer'>
+   <title>Build Container</title>
+<programlisting>
+buildContainer {
+  cmd = with pkgs; writeScript "run.sh" ''
+    #!${bash}/bin/bash
+    ${coreutils}/bin/exec ${bash}/bin/bash
+  ''; <co xml:id='ex-ociTools-buildContainer-1' />
+
+  mounts = [
+    { destination = "/data";
+      type = "none";
+      source = "/var/my_app";
+      options = [ "bind" ]; }
+  ]; <co xml:id='ex-ociTools-buildContainer-2' />
+
+  packages = with pkgs; [ curl git vim ]; <co xml:id='ex-ociTools-buildContainer-3' />
+
+  os = "linux"; <co xml:id='ex-ociTools-buildContainer-4' />
+
+  arch = "x86_64"; <co xml:id='ex-ociTools-buildContainer-5' />
+
+  readonly = false; <co xml:id='ex-ociTools-buildContainer-6' />
+}
+
+    </programlisting>
+   <calloutlist>
+    <callout arearefs='ex-ociTools-buildContainer-1'>
+     <para>
+      <varname>cmd</varname> specifies the program to run inside the container.
+      This is the only required argument for <varname>buildContainer</varname>.
+      All referenced packages inside the derivation will be made available
+      inside the container
+     </para>
+    </callout>
+    <callout arearefs='ex-ociTools-buildContainer-2'>
+     <para>
+      <varname>mounts</varname> specifies additional mount points chosen by the
+      user. By default only required mountpoints are made available inside the
+      container.
+     </para>
+    </callout>
+    <callout arearefs='ex-ociTools-buildContainer-3'>
+     <para>
+      <varname>packages</varname> is a list of additional packages linked
+      inside the container. Their outputs will be merged into a single
+      derivation in the <varname>PATH</varname>.
+     </para>
+    </callout>
+    <callout arearefs='ex-ociTools-buildContainer-4'>
+     <para>
+      <varname>os</varname> can be used to change the OS string for a
+      container. By default it is <literal>linux</literal> and can be omitted.
+     </para>
+    </callout>
+    <callout arearefs='ex-ociTools-buildContainer-5'>
+     <para>
+      <varname>arch</varname> can be used to change the computer architecture
+      string for a container. By default it's <literal>x86_64</literal> and can
+      be omitted.
```

Comment:
```
<varname>arch</varname> changes the computer architecture string for a container.
The default value is <literal>x86_64</literal>.
```

Reply:
Same question as for the `os` attribute. How would one use a foreign different architecture
and should not the packages then also come from the same architecture? 
I would rather default to the OS/arch of the `hostPlatform` to implement proper cross-compiled OCI images instead of letting the user to choose one.